### PR TITLE
Updating Team Analytics, splitting out queue time and FRT

### PR DESCRIPTION
**Pull request by deepwired**
Base restructuring commit

### CODE REVIEW

File: docs/agent-chat/analytics.md [hunk 1]
```diff
@@ -2,19 +2,43 @@
...
 
-1. **Avg. First Response Time** (FRT)
+1. **Average Agent First Response Time** (FRT)
 
-    FRT is the average of first response times taken by the agent to send the first response to end users. Only the chats whose first response was sent during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). Queue time is included in this metric. As we check this from first user message timestamp to the first agent response timestamp.
-    
-2. **Avg. Response Time** (RT)
+    AFRT is the average of first response times taken by the agent to send the first response to end users after a chat was assigned to the agent. Only the chats which were initiated by users during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). As we check this from first user message timestamp to the first agent response timestamp.
+
+2. **Average Queue Time** (RT)
```
Comment (medhaanand) [inline docs/agent-chat/analytics.md:32]: I see in the Figma design that with queue time and without queue time will hold true for all Time stats. I think we should explain all these three matrices separately with queue time and without queue time.

File: docs/agent-chat/analytics.md [hunk 2]
```diff
@@ -2,19 +2,43 @@
...
+
+2. **Total Number of agents**
+
+    The value here gives us the number of agents, who received atleast a single chat in the Team. Only those agents who received any chat(s) are counted. Also note that an agent could be part of multiple teams. 
+
+3. **Time saved of agents**
+
+    All chats where an Haptik bot (gogo) took part, we add up the individual resolution time for all such chats, that would be the total amount of time, Haptik bot (gogo) was responding to the end user, instead of an human agent. 
+    
+> This metric is calculated on the `business` level i.e. we show the "Time saved of agents" value totalled for all teams in a business (for the chosen Team). Since, most of our clients have one team per business, this apporach works.
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:22]: @bhaveshpatelh i can remove this right?

Comment (deepwired) [reply docs/agent-chat/analytics.md:22]: as in, we are currently showing time saved by gogo for a team only...

File: docs/agent-chat/analytics.md (hunk 1 above)
Comment (deepwired) [reply docs/agent-chat/analytics.md:31]: https://www.figma.com/file/iIKTe6BIMmokH3ELQRQxjl/Athena-Analytics?node-id=32%3A0

this is the design....

File: docs/agent-chat/analytics.md (hunk 2 above)
Comment (bhaveshpatelh) [reply docs/agent-chat/analytics.md]: Yeah, This is not relevant anymore, we can remove it.

File: docs/agent-chat/analytics.md [hunk 3]
```diff
@@ -2,21 +2,51 @@
...
+
+
 ## Time Stats
 
 ![Time_stats](assets/time_stats.png)
 
-1. **Avg. First Response Time** (FRT)
+1. **Average Agent First Response Time** (FRT)
+
+    AFRT is the average of first response times taken by the agent to send the first response to end users after a chat was assigned to the agent. Only the chats which were initiated by users during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). As we check this from chat assignment to agent till his first response to the chat.
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:29]:
```suggestion
    Average agent first response time is the average of first response times taken by the agent to send the first response to end users after a chat was assigned to the agent. Only the chats which were initiated by users during the selected time range will be taken into account. 
> Note: This metric does not include Queue Time for the chats
```

File: docs/agent-chat/analytics.md [hunk 4]
```diff
@@ -2,21 +2,51 @@
...
+
+2. **Average Queue Time** (RT)
 
-    FRT is the average of first response times taken by the agent to send the first response to end users. Only the chats whose first response was sent during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). Queue time is included in this metric. As we check this from first user message timestamp to the first agent response timestamp.
+    AQT is the average of total amount of time, chats have stayed within a queue. Chats remain in queue for the below mentioned reasons
     
-2. **Avg. Response Time** (RT)
+    1. A fresh chat has been initiated for a team, and there are no free agent slots avaiable.
+    2. A chat has been re-assigned to an Agent who is currently at max concurrency.
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:36]:
```suggestion
    2. A chat has been re-assigned to an Agent who is currently at max concurrency.
    3. A chat was put in waiting state, when the user replied the chat got allocated back to the agent queue, but the agent was again at max concurrency.
```

File: docs/agent-chat/analytics.md [hunk 5]
```diff
@@ -2,21 +2,51 @@
...
+    
+    In both of the scenarios mentioned above, the chat will remain in queue untill an agent receives the chat.
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:38]:
```suggestion
    In all of the scenarios mentioned above, the chat will remain in queue untill an agent receives the chat. 
```

File: docs/agent-chat/analytics.md [hunk 6]
```diff
@@ -2,21 +2,51 @@
...
-    FRT is the average of first response times taken by the agent to send the first response to end users. Only the chats whose first response was sent during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). Queue time is included in this metric. As we check this from first user message timestamp to the first agent response timestamp.
+    AQT is the average of total amount of time, chats have stayed within a queue. Chats remain in queue for the below mentioned reasons
     
-2. **Avg. Response Time** (RT)
+    1. A fresh chat has been initiated for a team, and there are no free agent slots avaiable.
+    2. A chat has been re-assigned to an Agent who is currently at max concurrency.
+    
+    In both of the scenarios mentioned above, the chat will remain in queue untill an agent receives the chat. 
+    
+> Number of free slots per agent depends on the number of chats currently assigned to the agent and chat concurrency set for the agent.
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:40]:
```suggestion
> Number of free slots per agent is the difference between chat concurrency set for the agent and the number of chats currently assigned to the agent.
```

File: docs/agent-chat/analytics.md [hunk 7]
```diff
@@ -2,21 +2,51 @@
...
+
+
+3. **Average Response Time** (RT)
 
-    RT calculation is the average time taken by the agent to respond in a chat to all messages and not just the first message. Only the chats whose responses were sent during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). For the user’s 2nd, 3rd and so on messages, the response time is simply the delta in the user message timestamp and the agent’s response timestamp. 
+    RT calculation is the average time taken by the agent to respond in a chat to all messages and not just the first message. Only the chats which had messages exchanged during the selected time range will be taken into account. In other words, average time taken by an agent to consecutively respond to a single or multiple user messages in cronological order.
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:45]:
```suggestion
    RT calculation is the average time taken by the agent to respond in a chat to all messages and not just the first message. Only the chats which were initiated during the selected time range will be taken into account. In other words, the average time taken by an agent to consecutively respond to a single or multiple user messages in chronological order.
```

File: docs/agent-chat/analytics.md [hunk 8]
```diff
@@ -2,21 +2,51 @@
 title: Analytics
 ---
 
+
+## Overview
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:6]: all below metrics are based out of chat initiation time, within the selected time range

File: docs/agent-chat/analytics.md [hunk 9]
```diff
@@ -2,21 +2,51 @@
...
     
-3. **Avg. Resolution Time**
+4. **Average Agent Resolution Time**
 
-    This metric 'Average resolution time' should be the average time taken by the agent till there are no messages sent in a chat session. Only the chats that were completed during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). Resolution time is simply calculated from the user’s first message sent timestamp till the last agent/user message timestamp in a chat.
+    This metric 'Average resolution time' should be the average time taken by the agent from the time when the chat was received till it was completed, re-assigned to marked as waiting. Only the chats that were completed during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). Agent Resolution time is simply calculated from the user’s first message sent timestamp till the last agent/user message timestamp in a chat, without queue time.
```
Comment (deepwired) [inline docs/agent-chat/analytics.md:49]:
```suggestion
    This metric 'Average agent resolution time' should be the average time taken by the agent from the time when the chat was received till it was completed, re-assigned to, or marked as waiting. Only the chats that were initiated during the selected time range will be taken into account. Agent Resolution time is simply calculated from the user’s first message sent timestamp till the last agent/user message timestamp in a chat, without queue time.
```

**Review by medhaanand — APPROVED** [review]

**Pull request merged**